test/e2e/performanceprofile/functests/0_config/config.go

```diff
@@ -158,9 +164,18 @@ func externalPerformanceProfile(performanceManifest string) (*performancev2.Perf
 	return profile, nil
 }
 
-func testProfile() *performancev2.PerformanceProfile {
-	reserved := performancev2.CPUSet("0")
-	isolated := performancev2.CPUSet("1-3")
+func testProfile() (*performancev2.PerformanceProfile, error) {
+	//Get cpus capacity on a node; typically it is same for all nodes
+	workerNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+	Expect(err).ToNot(HaveOccurred())
+	capacityCPU, _ := workerNodes[0].Status.Capacity.Cpu().AsInt64()
+	//consider cpu siblings - avoid noisy neighbor case
+	if capacityCPU < 4 {
+		//should never reach but still
+		return nil, fmt.Errorf("insufficient cpus on a node to create a valid performanceprofile for the tests, found %d cpus, expects at least 4", capacityCPU)
+	}
+	reserved := performancev2.CPUSet(fmt.Sprintf("0,%d", capacityCPU/2))
+	isolated := performancev2.CPUSet(fmt.Sprintf("1-%d,%d-%d", capacityCPU/2-1, capacityCPU/2+1, capacityCPU-1))
```

Comment:
Are 2 reserved CPUs are enough for machine with a big amount of cpus?

Reply:
This will cause issues on BM because on HT enabled if we don't pick cpus from the same core.

Reply:
Also the same for isolated, In terms of upstream where there are only 4 cpus, No Numa, this works , but as we use this on multinuma systems with HT enabled, this gets complicated. I strongly suggest to use PPC . Otherwise you will have to duplicate what ppc does here.

Reply:
Using PPC can work here, even though it will make the lane setup significantly more complex. But I agree we're reimplementing PPC.
The best IMO would be reuse part of PPC code, but we would need more detailed HW information to enable PPC.